compiler/rustc_hir_typeck/src/mem_categorization.rs

```diff
@@ -136,35 +127,38 @@ impl<'a, 'tcx> MemCategorizationContext<'a, 'tcx> {
                     Err(())
                 } else if ty.is_ty_var() {
                     debug!("resolve_type_vars_or_error: infer var from {:?}", ty);
-                    self.tcx()
+                    self.tcx
                         .dcx()
-                        .span_delayed_bug(self.tcx().hir().span(id), "encountered type variable");
+                        .span_delayed_bug(self.tcx.hir().span(id), "encountered type variable");
                     Err(())
                 } else {
                     Ok(ty)
                 }
             }
             // FIXME
```

Comment:
existing: what's that `FIXME`?

Reply:
Probably due to not wanting to rely on `InferCtxt` taintedness for expr-use-analysis.